cmd/statusd/debug/commands.go

```diff
@@ -0,0 +1,175 @@
+package debug
+
+import (
+	"fmt"
+	"go/ast"
+	"go/parser"
+	"go/token"
+	"reflect"
+	"strconv"
+	"strings"
+
+	"github.com/status-im/status-go/geth/api"
+	"github.com/status-im/status-go/geth/common"
+	"github.com/status-im/status-go/geth/params"
+)
+
+// command contains the result of a parsed command line and
+// is able to execute it on the command set.
+type command struct {
+	funcName string
+	args     []interface{}
+}
+
+// newCommand parses a command line and returns the command
+// for further usage.
+func newCommand(commandLine string) (*command, error) {
+	expr, err := parser.ParseExpr(commandLine)
+	if err != nil {
+		return nil, err
+	}
+	switch expr := expr.(type) {
+	case *ast.CallExpr:
+		return &command{
+			funcName: (expr.Fun.(*ast.Ident)).Name,
```

Comment:
This line has a potential to explode.

Reply:
Could you explain more?

Reply:
Type casting without checking if it's ok. Unless `expr.Fun.(*ast.Ident)` is always possible. I haven't worked with `ast`.

Reply:
I second that. `ast.CallExpr.Fun` is an `Expr` interface and can be of a few concrete types, not only `*ast.Ident`. Unless there is some preliminary validation happens, (invalid) command like `node.Start()` will result in panic here as `Fun` will be of type `*ast.SelectorExpr` instead.

@themue I think, simple comma-ok interface type casting would do the job. Also, this method is a good candidate for unit testing.

Reply:
Done. Sadly in tests not have been able to produce a non-Ident Expr.